### PR TITLE
[BugFix] Fix Qwen3-Omni CUDA graph capture lifecycle

### LINKED ISSUE #515: Qwen3-Omni MMSU accuracy drifts under thinker CUDA graph batching
## Summary

MMSU accuracy for Qwen3-Omni drifts downward when the thinker runs batched requests through the CUDA graph path. The effect is large enough to explain why the fused MMSU config can either preserve the new #477 accuracy threshold or restore the old QPS target, but not both at the same time.

This appears independent of MMSU parsing, request failures, router routing, and the audio encoder. The issue is isolated to the Qwen3-Omni thinker AR path under batched CUDA graph execution.

## Reproduction Context

Hardware/runtime:

- Hardware platform: H200 GPU
- Model: `Qwen/Qwen3-Omni-30B-A3B-Instruct`
- Dataset: `zhaochenyang20/mmsu-ci-2000`
- Workload: MMSU audio-in / text-out, `modalities=text`
- Client concurrency: 8 for the single-worker ablations

Server shape:

- Fused text-path MMSU config:
  - `preprocessing + image_encoder + audio_encoder + mm_aggregate + thinker + decode`
  - no `talker_ar`
  - no `code2wav`

## Results

| Config | Accuracy | Correct | QPS | Mean latency |
| --- | ---: | ---: | ---: | ---: |
| fused single worker, `max_running_requests=1`, CUDA graph on | 0.7080 | 1416/2000 | 17.72 | 0.451s |
| fused single worker, `max_running_requests=4`, CUDA graph on | 0.6940 | 1388/2000 | 31.993 | 0.250s |
| fused single worker, `max_running_requests=4`, CUDA graph off | 0.7060 | 1412/2000 | 19.008 | 0.420s |
| fused single worker, `max_running_requests=4`, CUDA graph on, graph padding disabled | 0.7015 | 1403/2000 | 31.416 | 0.254s |

The same #477 merge commit also appears too close to the 0.7065 threshold:

| Config | Accuracy | Correct | QPS | Mean latency |
| --- | ---: | ---: | ---: | ---: |
| #477 merge commit, router DP2 colocated topology, concurrency 16 | 0.7045 | 1409/2000 | 17.13 | 0.931s |

## Per-sample Diff

Against the `max_running_requests=1` baseline:

- `max_running_requests=4`, CUDA graph on:
  - 126 predicted-choice changes
  - 62 lost correct
  - 34 gained correct
  - net -28 correct
- `max_running_requests=4`, CUDA graph off:
  - 109 predicted-choice changes
  - 43 lost correct
  - 39 gained correct
  - net -4 correct
- CUDA graph off repaired 32 samples that were correct in the `max_running_requests=1` baseline but wrong in the CUDA graph batch run.

Examples repaired by disabling thinker CUDA graph:

| Task | Sample ID | Gold | max=1 | max=4 graph on | max=4 graph off |
| --- | --- | --- | --- | --- | --- |
| accent_identification | `accent_identification_c3618122-1dfc-44f1-a848-e8a81bd0f499` | B | B | A | B |
| emotion_recognition | `emotion_recognition_b664a17f-7d7b-421c-b4b0-b9008563cbbe` | D | D | C | D |
| intent_detection | `intent_detection_3e9c8199-01d5-43c2-bfbd-c24429c96160` | C | C | D | C |
| speech_duration_estimation | `speech_duration_estimation_3510cf2e-0323-4950-996b-5b6c8880adbf` | D | D | C | D |

## Current Hypothesis

The main accuracy loss is caused by the thinker CUDA graph batch execution path. Disabling CUDA graph padding recovers part of the loss while preserving most of the throughput, which suggests the graph batch shape/padding path contributes to the drift, but padding alone does not explain all of it.

There is still a small residual difference between `max_running_requests=1` and `max_running_requests=4` with CUDA graph disabled, likely from ordinary batched-kernel numerical differences or scheduling nondeterminism. The larger regression is specific to the CUDA graph path.

## Suggested Next Step

Run a logits-level probe for the same MMSU samples under:

- eager vs CUDA graph
- batch size 1/2/4
- CUDA graph padding on/off

For each condition, compare final logits for the answer choices A/B/C/D before decoding. This should identify whether the drift comes from decode attention, MoE routing/kernel behavior, graph padding/replay state, or another captured-graph side effect.

Related PR: #514

### PR DESCRIPTION
**Pull request by Hayden727**
Closes #515.

## Summary

- Fix Qwen3-Omni thinker CUDA graph lifecycle so the text-only path relies on upstream SGLang's automatic `ModelRunner` graph capture and skips the redundant bootstrap `init_device_graphs()` call.
- Keep the existing deferred capture behavior only for the speech/hidden-state path, where hidden capture hooks are installed after worker construction and must be present before graph capture.
- Add parameterized unit coverage for both paths: text-only uses upstream capture only; speech defers and manually captures once after hidden hooks are installed.

## Root Cause

The Qwen3-Omni thinker text-only path started with `disable_cuda_graph=False`, so upstream SGLang captured CUDA graphs during model worker construction. The Omni bootstrap then unconditionally called `model_worker.model_runner.init_device_graphs()` when CUDA graph was requested, causing the text-only thinker to capture twice. This matched the issue's batch-sensitive CUDA graph drift symptoms.

## Validation

Final commit validation:

- `python -m pytest -q tests/unit_test/qwen3_omni/test_pipeline.py::test_qwen_thinker_cuda_graph_capture_lifecycle tests/unit_test/qwen3_omni/test_config_manager.py::test_qwen3_omni_mmsu_example_config_uses_text_pipeline`
  - Environment: H200 GPU platform
  - Result: `3 passed, 2 warnings in 6.92s`
- Broader server-side unit sweep:
  - `python -m pytest -q tests/unit_test/qwen3_omni/test_pipeline.py tests/unit_test/qwen3_omni/test_config_manager.py tests/unit_test/qwen3_omni/test_cli.py tests/unit_test/qwen3_omni/test_sglang_ar_budget.py`
  - Result: `72 passed, 2 warnings in 6.94s`
- Commit hooks: passed (`autoflake`, `isort`, `black-jupyter`, AST/check hooks, etc.)

H200 full MMSU validation while isolating the one-capture fix path:

- Environment: H200 GPU platform, Qwen/Qwen3-Omni-30B-A3B-Instruct, text-only MMSU config, 2000 samples, temperature 0.0
- `max_concurrency=4`: `1405/2000 = 70.25%`
- `max_concurrency=16`: `1404/2000 = 70.20%`
- repeated `max_concurrency=4`: `1404/2000 = 70.20%`
- repeated `c=4` vs `c=16`: `diff_count=0` on per-sample parsed outputs
- first `c=4` vs later runs: `diff_count=7`, net 1 sample, consistent with small run-to-run nondeterminism rather than the issue's graph-on batch-size accuracy drop

Final commit H200 full MMSU validation:

- Commit: `0e625c9`
- Environment: H200 GPU platform, Qwen/Qwen3-Omni-30B-A3B-Instruct, text-only MMSU config, 2000 samples, temperature 0.0
- CUDA graph startup log: one upstream capture sequence for `bs=4/2/1`; no second bootstrap capture
- `max_concurrency=4`: `1407/2000 = 70.35%`, throughput `19.55 req/s`, mean latency `0.204s`, p95 `0.217s`
- `max_concurrency=16`: `1404/2000 = 70.20%`, throughput `35.72 req/s`, mean latency `0.446s`, p95 `0.605s`
- repeated `max_concurrency=4`: `1404/2000 = 70.20%`, throughput `22.53 req/s`, mean latency `0.177s`, p95 `0.207s`
- repeated `c=4` vs `c=16`: `diff_count=0`, same per-sample parsed outputs
- first `c=4` vs repeated `c=4`: `diff_count=15`, net -3 correct, indicating small run-to-run nondeterminism rather than a batch-size-specific CUDA graph regression

### CODE REVIEW

**Comment by Hayden727** [comment]
Additional server validation for commit `0e625c9`:

- Hardware platform: H200 GPU
- Command:
  `python -m pytest -q tests/unit_test/qwen3_omni/test_pipeline.py tests/unit_test/qwen3_omni/test_config_manager.py tests/unit_test/qwen3_omni/test_cli.py tests/unit_test/qwen3_omni/test_sglang_ar_budget.py`
- Result: `72 passed, 2 warnings in 6.94s`

Final-commit full MMSU rerun is now complete on H200:

- Startup log showed a single upstream CUDA graph capture sequence for `bs=4/2/1`, with no second bootstrap capture.
- `max_concurrency=4`: `1407/2000 = 70.35%`, throughput `19.55 req/s`, mean latency `0.204s`, p95 `0.217s`.
- `max_concurrency=16`: `1404/2000 = 70.20%`, throughput `35.72 req/s`, mean latency `0.446s`, p95 `0.605s`.
- repeated `max_concurrency=4`: `1404/2000 = 70.20%`, throughput `22.53 req/s`, mean latency `0.177s`, p95 `0.207s`.
- repeated `c=4` vs `c=16`: `diff_count=0`, same per-sample parsed outputs.
- first `c=4` vs repeated `c=4`: `diff_count=15`, net -3 correct, consistent with small run-to-run nondeterminism rather than a batch-size-specific CUDA graph regression.

So the final commit has now passed broader server-side unit validation and full H200 MMSU validation. The batch-size accuracy drop reported in #515 does not reproduce with the redundant text-only bootstrap capture removed.

**Review by Ratish1 — APPROVED** [review]
Nice Fix @Hayden727 . LGTM
